### PR TITLE
New Port - PowerPutt Legend

### PR DESCRIPTION
**Pull request by steventrev**
# New Port for PowerPutt Legend

## Game Information
- **Title**: PowerPutt Legend
- **URL**: https://team-bugulon.itch.io/powerputt-legend

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [x] 480x320
- [x] 640x480
- [x] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

File: ports/powerputtlegend/gameinfo.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<gameList>
+  <game>
+    <path>./Powerputt Legend.sh</path>
+    <name>Powerputt Legend</name>
+    <desc>PowerPutt Legend is a unique minigolf game with RPG elements, meant to look and feel like it's straight out of the NES. Slay monsters, level up, and unlock a series of items to improve your strategy. </desc>
+    <image>./PowerPuttLegend/screenshot.jpg</image>
```
Comment (kloptops) [inline ports/powerputtlegend/gameinfo.xml:7]: Wrong case in path.

```suggestion
    <image>./powerputtlegend/screenshot.jpg</image>
```

**Review by kloptops — CHANGES_REQUESTED** [review]
Small change otherwise looks good. Is it `PowerPutt Legend` or `Powerputt Legend`?

It's a bit inconsistent in the `port.json` and `gameinfo.xml`. :)

File: ports/powerputtlegend/port.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "genres": [
+      "adventure"
+    ],
+    "image": null,
+    "rtr": true,
+    "exp": false,
+    "runtime": null,
+    "reqs": [],
+    "arch": [
+      "armhf"
```
Comment (JeodC) [inline ports/powerputtlegend/port.json:25]: This is using gmloadernext, so it's aarch64.

File: ports/powerputtlegend/cover.jpg
Comment (JeodC) [inline ports/powerputtlegend/cover.jpg:1]: I think images need to be .png format, but this could be wrong.

**Review by JeodC — CHANGES_REQUESTED** [review]

File: ports/powerputtlegend/cover.jpg
Comment (steventrev) [reply ports/powerputtlegend/cover.jpg:1]: Happy to change, but should also modify PortMaster-New readme:

Comment (steventrev) [reply ports/powerputtlegend/cover.jpg:1]: I can change, would also need revision of sources that reference cover.jpg:

- https://github.com/PortsMaster/PortMaster-New?tab=readme-ov-file#new-port-structure

- https://portmaster.games/packaging.html

**Review by JeodC — APPROVED** [review]

File: ports/powerputtlegend/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+## Notes
+
+Thanks [Team Bugulon](https://team-bugulon.itch.io/) for creating this game and allowing us to distribute it!
```
Comment (Cebion) [inline ports/powerputtlegend/README.md:3]: Missing URL to the Game.

File: ports/powerputtlegend/port.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "porter": [
+      "coolz"
+    ],
+    "desc": "PowerPutt Legend is a unique minigolf game with RPG elements, meant to look and feel like it's straight out of the NES. Slay monsters, level up, and unlock a series of items to improve your strategy.",
+    "inst": "Ready to run! Thanks to Team Bugulon granting us permission to distribute.",
+    "genres": [
+      "adventure"
+    ],
+    "image": null,
+    "rtr": true,
```
Comment (Cebion) [inline ports/powerputtlegend/port.json:20]: rtr:true means gamedata is included.
If that is the case a license needs also to be provided.

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/powerputtlegend/Powerputt Legend.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
+
+# Display loading splash
+$ESUDO ./libs/splash "splash.png" 1 
+$ESUDO ./libs/splash "splash.png" 1000
+
+# Run game
+$GPTOKEYB "gmloadernext" &
+pm_platform_helper "$GAMEDIR/gmloadernext"
+./gmloadernext game.apk
```
Comment (cdeletre) [inline ports/powerputtlegend/Powerputt Legend.sh:38]: gmloadernext doesn't use the command line argument, `game.apk` can be remove in this line.
```suggestion
./gmloadernext
```

File: ports/powerputtlegend/Powerputt Legend.sh [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+cd $GAMEDIR
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+$ESUDO chmod +x -R $GAMEDIR/*
+
+# Exports
+export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/lib:$GAMEDIR/libs:$LD_LIBRARY_PATH"
+export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
+
+# Display loading splash
+$ESUDO ./libs/splash "splash.png" 1
```
Comment (cdeletre) [inline ports/powerputtlegend/Powerputt Legend.sh:32]: This is a workaround for muOS so it's better to use it only when muOS is detected
```suggestion
[ "$CFW_NAME" == "muOS" ] && splash "splash.png" 1 # workaround for muOS
```

File: ports/powerputtlegend/gameinfo.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<gameList>
+  <game>
+    <path>./PowerPutt Legend.sh</path>
```
[the remark this answers is not in the document]
Comment (steventrev) [reply ports/powerputtlegend/gameinfo.xml:4]: Thanks! Casing needed explicit `git mv "Powerputt Legend.sh" "PowerPutt Legend.sh"` to be recognized. Probably Windows related.

File: ports/powerputtlegend/Powerputt Legend.sh [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+# Variables
+GAMEDIR="/$directory/ports/powerputtlegend"
+
+# CD and set permissions
+cd $GAMEDIR
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+$ESUDO chmod +x -R $GAMEDIR/*
+
+# Exports
+export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/lib:$GAMEDIR/libs:$LD_LIBRARY_PATH"
```
[the remark this answers is not in the document]
Comment (steventrev) [reply ports/powerputtlegend/Powerputt Legend.sh:28]: Libs folder added to support ArkOS in [c8c5d6e](https://github.com/PortsMaster/PortMaster-New/pull/782/commits/c8c5d6e3e518997d495a0401143bfc458bbbcd8d).

**Review by cdeletre — APPROVED** [review]

File: ports/powerputtlegend/gameinfo.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (steventrev) [reply ports/powerputtlegend/gameinfo.xml:4]: It appears this casing change still did not take on the latest commit. Will try again.

**Review by cdeletre — DISMISSED** [review]

File: ports/powerputtlegend/20240307_TeamBugulon-DistributionAgreement.png
Comment (cdeletre) [inline ports/powerputtlegend/20240307_TeamBugulon-DistributionAgreement.png:1]: This file goes into `ports/powerputtlegend/powerputtlegend/`

**Review by cdeletre — CHANGES_REQUESTED** [review]

**Review by cdeletre — DISMISSED** [review]

**Review by JeodC — APPROVED** [review]

**Pull request merged**